### PR TITLE
manifest: nrf_modem: v2.1.0

### PR DESCRIPTION
**Pull request by lemrey**
Updates for Modem library v2.1.0.

See the changelog in nrfxlib for a detailed list of changes
in this Modem library release.

This patch updates the nrf_modem_lib_trace component
to use the newly introduced `nrf_modem_trace_get()` function
to retrieve traces. Traces are printed by a thread
that is automatically started when traces are enabled.

test-sdk-nrf: sdk-nrf-libmodem-tracing
test-fw-nrfconnect-nrf_iot: mosh_libmodem_kconfig_change

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

File: doc/nrf/libraries/modem/nrf_modem_lib.rst [hunk 1]
```diff
@@ -80,32 +80,18 @@ In this case, the characteristics of the allocations made by these functions dep
...
-This is done automatically when using the OS abstraction layer.
-
-If the application wants to stop an ongoing trace session, it can use the :c:func:`nrf_modem_lib_trace_stop` function.
-The :c:func:`nrf_modem_lib_trace_start` function supports activating a subset of traces or all traces.
-
-Thread-based processing
-=======================
+If the application wants to set a trace level and start/resume a trace session, it can use the :c:func:`nrf_modem_lib_trace_level_set` function.
+If the application wants to pause an ongoing trace session, it can use the :c:func:`nrf_modem_lib_trace_pause` function.
+If the application wants to resume an ongoing trace session, it can use the :c:func:`nrf_modem_lib_trace_resume` function.
```
Comment (divipillai) [inline doc/nrf/libraries/modem/nrf_modem_lib.rst:92]:
```suggestion
The application can use the following functions for different tracing activities:

* :c:func:`nrf_modem_lib_trace_level_set` - This function sets a trace level and starts or resumes a trace session.
* :c:func:`nrf_modem_lib_trace_pause` - This function is used to pause an ongoing trace session.
* :c:func:`nrf_modem_lib_trace_resume` - This function resumes an ongoing trace session.

```

**Review by divipillai — COMMENTED** [review]
Some minor updates.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -417,6 +417,21 @@ Modem libraries
...
+
+    * Removed:
+
+      * The following Kconfig options have been removed:
+
+        * ``CONFIG_NRF_MODEM_LIB_TRACE_THREAD_PROCESSING``
+        * ``CONFIG_NRF_MODEM_LIB_TRACE_HEAP_SIZE``
+        * ``CONFIG_NRF_MODEM_LIB_TRACE_HEAP_SIZE_OVERRIDE``
+        * ``CONFIG_NRF_MODEM_LIB_TRACE_HEAP_DUMP_PERIODIC``
+        * ``CONFIG_NRF_MODEM_LIB_TRACE_HEAP_DUMP_PERIOD_MS``
```
Comment (divipillai) [inline doc/nrf/releases/release-notes-changelog.rst:434]:
```suggestion
      * The modem trace module to use the new APIs in Modem library.
        The modem trace output is now handled by a dedicated thread that starts automatically.
        The trace thread is synchronized with the initialization and shutdown operations of the Modem library.
      * The Kconfig option ``CONFIG_NRF_MODEM_LIB_TRACE_ENABLED`` has been renamed to :kconfig:option:`CONFIG_NRF_MODEM_LIB_TRACE`.

    * Removed:

      * The following Kconfig options:

        * ``CONFIG_NRF_MODEM_LIB_TRACE_THREAD_PROCESSING``
        * ``CONFIG_NRF_MODEM_LIB_TRACE_HEAP_SIZE``
        * ``CONFIG_NRF_MODEM_LIB_TRACE_HEAP_SIZE_OVERRIDE``
        * ``CONFIG_NRF_MODEM_LIB_TRACE_HEAP_DUMP_PERIODIC``
        * ``CONFIG_NRF_MODEM_LIB_TRACE_HEAP_DUMP_PERIOD_MS``
```

Comment (lemrey) [reply doc/nrf/releases/release-notes-changelog.rst]: Thank you, updated

**Review by divipillai — APPROVED** [review]

File: lib/nrf_modem_lib/Kconfig.modemlib [hunk 3]
```diff
@@ -31,56 +31,52 @@ config NRF_MODEM_LIB_TRACE_ENABLED
...
+if NRF_MODEM_LIB_TRACE
 
-choice NRF_MODEM_LIB_TRACE
+config NRF_MODEM_LIB_TRACE_LEVEL_OVERRIDE
+	bool "Set default trace level upon start"
+	default y
+
+if NRF_MODEM_LIB_TRACE_LEVEL_OVERRIDE
+
+choice
```
Comment (eivindj-nordic) [inline lib/nrf_modem_lib/Kconfig.modemlib:42]: Looks like the choice needs a name to compile. Obvious name is `NRF_MODEM_LIB_TRACE_LEVEL`. However that is taken by the integer config below. Maybe rename the integer config to `NRF_MODEM_LIB_TRACE_LEVEL_INT`? This is never set by the application, but depends solely on the choice of the trace level. Then the choice can be named `NRF_MODEM_LIB_TRACE_LEVEL`.

File: tests/lib/nrf_modem_lib/nrf_modem_lib_trace/prj.conf [hunk 4]
```diff
@@ -5,6 +5,7 @@
 #
 
 CONFIG_UNITY=y
-CONFIG_ASSERT=y
-CONFIG_NRF_MODEM_LIB_TRACE_ENABLED=y
-CONFIG_NRF_MODEM_LIB_TRACE_THREAD_PROCESSING=y
+CONFIG_ASSERT=n
+CONFIG_NRF_MODEM_LIB_TRACE=y
```
Comment (eivindj-nordic) [inline tests/lib/nrf_modem_lib/nrf_modem_lib_trace/prj.conf:9]: Either add `CONFIG_NRF_MODEM_LIB_TRACE_LEVEL_OVERRIDE=n` or rewrite tests to expect setting the trace level at initialization.

File: west.yml [hunk 5]
```diff
@@ -112,7 +112,7 @@ manifest:
     - name: nrfxlib
       repo-path: sdk-nrfxlib
       path: nrfxlib
-      revision: 5e9165e0f460fb6b2fe30d85a88dc74e5ad98320
+      revision: pull/783/head
```
Comment (rlubos) [inline west.yml:115]: Please update SHA: `9dba61fe2328da8bba1a432f285d8ddf259f517b`

Comment (lemrey) [reply west.yml]: Thanks

**Review by rlubos — APPROVED** [review]

**Pull request merged**